src/base/bittorrent/peer_filter_session_plugin.hpp

```diff
@@ -49,33 +50,45 @@ class peer_filter_session_plugin final : public lt::plugin
   {
   }
 
-  std::shared_ptr<lt::torrent_plugin> new_torrent(const lt::torrent_handle&, client_data) override
+  std::shared_ptr<lt::torrent_plugin> new_torrent(const lt::torrent_handle& th, client_data) override
   {
     // do not waste CPU and memory for useless objects when no filters are enabled
     if (!m_blacklist && !m_whitelist)
       return nullptr;
+
+    // ignore private torrents
+    if (th.torrent_file() && th.torrent_file()->priv())
+      return nullptr;
+
     return std::make_shared<peer_action_plugin>([this](auto&&... args) { return filter(args...); }, drop_peer_connection);
   }
 
 protected:
   bool filter(const lt::peer_info& info, bool handshake, bool* stop_filtering) const
   {
     if (m_blacklist) {
-      bool matched = m_blacklist->match_peer(info, false);
-      *stop_filtering = !handshake && !matched;
-      if (matched)
+      bool matched_blacklist = m_blacklist->match_peer(info, handshake);
+      if (matched_blacklist) {
         peer_logger_singleton::instance().log_peer(info, "blacklist");
-      return matched;
+        *stop_filtering = true;
+        return true;
+      }
     }
 
     if (m_whitelist) {
-      bool matched = m_whitelist->match_peer(info, handshake);
-      *stop_filtering = !handshake && matched;
-      if (!matched)
+      bool matched_whitelist = m_whitelist->match_peer(info, handshake);
+      if (!matched_whitelist) {
         peer_logger_singleton::instance().log_peer(info, "whitelist");
-      return !matched;
+        *stop_filtering = true;
+        return true;
+      }
     }
 
+    // if the peer got passed the handshake phase and get here,
+    // don't filter it anymore
```

Comment:
just a cosmetic suggestion - make this comment one-line. it will not look too long or odd, this function has longer lines

Reply:
Okay, will do that.